xep-0407.xml

```diff
@@ -198,19 +204,19 @@
         Two approaches to message retraction can be used.  In the first approach, the retracted message is simply removed.   This is appropriate where retraction is provided as a user service and the user has rights to remove messages sent from the record.
       </p>
       <p>
-        The second approach is to leave a tombstone, which if taken MUST be done in the following manner.  It is recommended to use a tombstone, as simply deleting the message may cause confusion with MAM queries.   Use of a tombstone is appropriate where it is desired to leave a record of the message that was redacted.
-        With this approach, the original message &lt;body&gt; is removed and replaced with a tombstone using the &lt;retracted&gt; element qualified by the 'urn:xmpp:mix:misc:0' namespace that shows the JID of user performing the retraction and the time of the retraction.
+        The second approach is to leave a tombstone.  It is recommended to use a tombstone, as simply deleting the message may cause confusion with MAM queries.   Use of a tombstone is appropriate where it is desired to leave a record of the message that was redacted. The tombstones are also defined in &xep0424;. A service which supports tombstones MUST advertise the 'urn:xmpp:message-retract:0#tombstone' feature in its Service Discovery responses.
       </p>
+
       <example caption="Retracted message tombstone in a MAM result"><![CDATA[
 <message id='aeb213' to='juliet@capulet.example/UUID-e3r/9264'>
   <result xmlns='urn:xmpp:mam:2' queryid='f27' id='28482-98726-73623'>
     <forwarded xmlns='urn:xmpp:forward:0'>
       <delay xmlns='urn:xmpp:delay' stamp='2010-07-10T23:08:25Z'/>
       <message xmlns='jabber:client' from="hag66@shakespeare.example"
                  to="macbeth@shakespeare.example">
-        <retracted xmlns='urn:xmpp:mix:misc:0' by='hag66@shakespeare.example'
-                 time='2010-07-10T23:08:25Z'/>
-      </message>
+        <retracted xmlns='urn:xmpp:message-retract:0' stamp='2010-07-10T23:08:25Z'>
+          <origin-id xmlns='urn:xmpp:sid:0' id='a568706e-867b-49f1-a31b-e428ba9cece1'/>
```

Comment:
This is missing the note of who retracted the message (also a comment from my original round ;))